File: app/src/main/java/com/sjk/deleterecentpictures/utils/CheckApkExist.java
package com.sjk.deleterecentpictures.utils;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.text.TextUtils;

public class CheckApkExist {
    public static boolean checkApkExist(Context context, String ApkPackageName) {
        if (TextUtils.isEmpty(ApkPackageName)) {
            return false;
        }
        try {
            ApplicationInfo info = context.getPackageManager().getApplicationInfo(ApkPackageName, PackageManager.GET_UNINSTALLED_PACKAGES);
            return true;
        } catch (PackageManager.NameNotFoundException e) {
            return false;
        }
    }
}

File: app/src/main/java/com/sjk/deleterecentpictures/utils/CheckApkExist.kt
package com.sjk.deleterecentpictures.utils

import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import android.text.TextUtils

object CheckApkExist {
    fun checkApkExist(context: Context?, ApkPackageName: String?): Boolean {
        return if (TextUtils.isEmpty(ApkPackageName)) {
            false
        } else try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                context?.packageManager?.getApplicationInfo(ApkPackageName!!, PackageManager.MATCH_UNINSTALLED_PACKAGES)
            } else {
                context?.packageManager?.getApplicationInfo(ApkPackageName!!, PackageManager.GET_UNINSTALLED_PACKAGES)
            }
            true
        } catch (e: PackageManager.NameNotFoundException) {
            false
        }
    }
}